client/src/main/kotlin/io/spine/examples/chatspn/desktop/Client.kt

```diff
@@ -0,0 +1,458 @@
+     *
+     * @param chat ID of the chat to observe messages in
+     * @param onUpdate will be called when a new chat message is posted,
+     *                     or when an existing message is updated
+     * @param onDelete will be called when a message is deleted
+     * @throws IllegalStateException if the user has not been authenticated
+     */
+    public fun observeMessages(
+        chat: ChatId,
+        onUpdate: (message: MessageView) -> Unit,
+        onDelete: (messageDeleted: MessageMarkedAsDeleted) -> Unit
+    ) {
+        checkNotNull(authenticatedUser) { "The user has not been authenticated" }
+        val updateSubscription = clientRequest()
+            .subscribeTo(MessageView::class.java)
+            .where(chat.stateFilter())
+            .observe(onUpdate)
+            .post()
+        val deletionSubscription = clientRequest()
+            .subscribeToEvent(MessageMarkedAsDeleted::class.java)
+            .where(chat.eventFilter())
+            .observe(onDelete)
+            .post()
+        messagesSubscriptions.add(updateSubscription)
+        messagesSubscriptions.add(deletionSubscription)
+    }
+
+    /**
+     * Stops messages observation.
+     */
+    public fun stopObservingMessages() {
+        messagesSubscriptions.forEach { subscription ->
+            client.subscriptions().cancel(subscription)
+        }
+        messagesSubscriptions.clear()
+    }
+
+    /**
+     * Provides `ClientRequest` on behalf of `authenticatedUser` if it exists,
+     * or as guest if it doesn't.
+     */
+    private fun clientRequest(): ClientRequest {
+        if (null == authenticatedUser) {
+            return client.asGuest()
+        }
+        return client.onBehalfOf(authenticatedUser!!.id)
+    }
+
+    /**
+     * Observes the provided event.
+     *
+     * @param id ID of the event state to observe
+     * @param event type of the event to observe
+     * @param onEmit will be called when the specified event emitted
+     * @return subscription object to cancel observation
+     */
+    private fun <E : EventMessage> observeEvent(
+        id: Message,
+        event: Class<E>,
+        onEmit: (event: E) -> Unit
+    ): Subscription {
+        val subscription = clientRequest()
+            .subscribeToEvent(event)
+            .where(EventFilter.eq(EventMessageField(Field.named("id")), id))
+            .observe(onEmit)
+            .post()
+        return subscription
+    }
+
+    /**
+     * Stops observation by provided subscription.
+     *
+     * @param subscription subscription to cancel
+     */
+    private fun stopObservation(subscription: Subscription) {
+        client.subscriptions()
+            .cancel(subscription)
+    }
+}
+
+/**
+ * Creates command to create an account for the user with `this` email.
+ *
+ * @param name name of the user to create an account for
+ * @return command to create an account
+ */
+private fun String.createAccountCommand(name: String): CreateAccount {
```

Comment:
I don't understand this at all.

If you would have just a single `String` sufficient in order to produce `CreateAccount`, that would make sense. But you need two.

Therefore, it should be an extension to `CreateAccount` instead.